File: src/main/java/no/nav/bidrag/beregn/forskudd/rest/controller/BeregnForskuddController.java
package no.nav.bidrag.beregn.forskudd.rest.controller;

import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.security.SecurityRequirement;
import no.nav.bidrag.beregn.forskudd.rest.dto.http.BeregnForskuddGrunnlag;
import no.nav.bidrag.beregn.forskudd.rest.dto.http.BeregnetForskuddResultat;
import no.nav.bidrag.beregn.forskudd.rest.service.BeregnForskuddService;
import no.nav.security.token.support.core.api.Protected;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/beregn")
@Protected
public class BeregnForskuddController {

  private final BeregnForskuddService beregnForskuddService;

  public BeregnForskuddController(BeregnForskuddService beregnForskuddService) {
    this.beregnForskuddService = beregnForskuddService;
  }

  @PostMapping(path = "/forskudd")
  @Operation(summary = "Beregner forskudd")
  @SecurityRequirement(name = "bearer-key")
  public ResponseEntity<BeregnetForskuddResultat> beregnForskudd(@RequestBody BeregnForskuddGrunnlag beregnForskuddGrunnlag) {
    var resultat = beregnForskuddService.beregn(beregnForskuddGrunnlag);
    return new ResponseEntity<>(resultat.getResponseEntity().getBody(), resultat.getResponseEntity().getStatusCode());
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/forskudd/rest/controller/BeregnForskuddController.kt
package no.nav.bidrag.beregn.forskudd.rest.controller

import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.security.SecurityRequirement
import io.swagger.v3.oas.models.examples.Example
import no.nav.bidrag.beregn.forskudd.BeregnForskuddApi
import no.nav.bidrag.commons.util.OpenApiExample
import no.nav.bidrag.transport.behandling.beregning.felles.BeregnGrunnlag
import no.nav.security.token.support.core.api.Protected
import org.springframework.context.annotation.Bean
import org.springframework.http.HttpMethod
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/beregn")
@Protected
class BeregnForskuddController(private val beregning: BeregnForskuddApi) {

    val eksempelfiler = (1..23).map { "forskudd_eksempel$it" }
    private fun createExample(filename: String): OpenApiExample {
        val example = Example()
        example.value = BeregnForskuddController::class.java.getResource("/testfiler/$filename.json")?.readText() ?: ""
        example.description = filename
        return OpenApiExample(
            example = example,
            path = "/beregn/forskudd",
            method = HttpMethod.POST,
        )
    }

    @Bean
    fun eksempler(): List<OpenApiExample> = eksempelfiler.map { createExample(it) }

    @PostMapping(path = ["/forskudd"])
    @Operation(summary = "Beregner forskudd")
    @SecurityRequirement(name = "bearer-key")
    fun beregnForskudd(@RequestBody beregnForskuddGrunnlag: BeregnGrunnlag) = beregning.beregn(beregnForskuddGrunnlag)
}
